falcon/routing/static.py

```diff
@@ -61,7 +71,8 @@ def __call__(self, req, resp):
 
         # NOTE(kgriffs): Check surrounding whitespace and strip trailing
         # periods, which are illegal on windows
-        if (not without_prefix or
+        # NOTE(CaselIT): An empty filename is allowed when fallback_filename is provided
```

Comment:
Let's say we mount a static route on `/static` from directory `/tmp/static/` with a fallback filename. If we do not allow an empty filename in the request, and the user asks for`/static`, the route would match, but we would send a 404 instead of the fallback filename.

By allowing the empty file, `file_path` would be set to `/tmp/static/.`, `open` fails since it `file_path` is a directory, so it will send the fallback file in the `except` block